### ISSUE TITLE
Saving of addons/game settings

### ISSUE THREAD

**Issue by Flamefire** [Return-To-The-Roots/s25client#587]
There was a bug, that addon settings are reset if you open the options w/o opening also the addons, which is now fixed in https://github.com/Return-To-The-Roots/s25client/commit/fd9af5f24ef79abdc475262db4dd0fb2f1448ead.
Although I think the fix would be better to save/load only in the addon window.

However game settings (speed etc) is never saved. Also addons are not saved when set when hosting a game. IMO it would be more user-friendly if the settings (addons and others) would be saved when the host starts a normal game (no lua file) or even if he just sets them. Then you can have your preferred settings and keep them every time.
ATM it is also not intuitive that the addon settings are saved for the options but not ingame although the dialog looks the same.

TLDR: When and what settings should be saved?

ccing @Flow86 @Spikeone @MarcusSt @jhkl

**Comment by Spikeone** [comment]
I actually like the way it works currently (without resetting) as I can set my "base always use those addons" in the options menu but changing addons for one game "special addon settings for this map" are not saved in the host screen. 
That way I can for example set limit 3 catapults always but I can set max rank for my current game, next time I start a game I would still have my 3 catapult limit but not the changed max rank.

I'd either leave it as it is (maybe some sort of hint?) or add a button for saving globally and all games.
